packages/plugins/type-designer/src/components/elements-type-container/columns.svelte

```diff
@@ -0,0 +1,188 @@
+    name: string;
+    visible: boolean;
+    items: DataTypeTemplates[];
+  };
+</script>
+
+<script lang="ts">
+  import Paper, { Content } from "@smui/paper";
+  import Button from "@smui/button";
+  import type { DataTypeTemplates } from "./data-type-templates";
+  import { createNewItem } from "./create-new-item";
+  import {
+    ELEMENT_NAMES,
+    type DataTypeTemplatesService,
+  } from "@oscd-plugins/core";
+  import ContentCard from "./content-card.svelte";
+  import IconWrapper from "@oscd-plugins/ui/src/components/icons/icon-wrapper.svelte";
+  import IconButton from "@smui/icon-button";
+
+  export let service: DataTypeTemplatesService;
+
+  const typeCluster = service.findTypeDesignerElements();
+
+  let columns: Column[] = [
+    {
+      name: ELEMENT_NAMES.substation,
+      visible: true,
+      items: typeCluster.substations,
+    },
+    {
+      name: ELEMENT_NAMES.voltageLevel,
+      visible: true,
+      items: typeCluster.voltageLevels,
+    },
+    { name: ELEMENT_NAMES.bay, visible: true, items: typeCluster.bays },
+    { name: ELEMENT_NAMES.ied, visible: true, items: typeCluster.ieds },
+    {
+      name: ELEMENT_NAMES.lDevice,
+      visible: true,
+      items: typeCluster.logicalDevices,
+    },
+    { name: ELEMENT_NAMES.lNode, visible: true, items: [] },
+  ];
+
+  function toggleColumnVisibility(index: number) {
+    columns = columns.map((column, i) =>
+      i === index ? { ...column, visible: !column.visible } : column,
+    );
+  }
+
+  function addItemToColumn(index: number) {
+    const column = columns[index];
+    const itemCount = column.items.length;
+    const newItem = createNewItem(column.name, itemCount);
+
+    if (newItem) {
+      columns = columns.map((col, i) =>
+        i === index ? { ...col, items: [...col.items, newItem] } : col,
+      );
+    }
+  }
+</script>
+
+<article class="columns-container" id="type-designer-columns">
+    {#each columns as column, index}
+      <Paper class="column">
+        <div class="column-header">
+          {#if column.visible}
+            <h2>{column.name}</h2>
+          {:else}
+            <h2>{column.name} (hidden)</h2>
+          {/if}
+          <IconButton on:click={() => toggleColumnVisibility(index)}>
+            <IconWrapper
+              icon={column.visible ? "visibility" : "visibility_off"}
+              fillColor="rgb(81, 159, 152)"
+            />
+          </IconButton>
+        </div>
+        {#if column.visible}
+          <Content class="content">
+            <div class="element-types">
+              <ContentCard elementType={column.items} />
+            </div>
+          </Content>
+        {/if}
+        <div class="add-button-container">
+          {#if column.visible && column.name !== ELEMENT_NAMES.lNode}
+            <Button class="add-button" on:click={() => addItemToColumn(index)}>
+              + Create {column.name}
+            </Button>
+          {/if}
+        </div>
+      </Paper>
+    {/each}
+</article>
```

Comment:
```suggestion
<div class="columns-container" id="type-designer-columns">
	{#each columns as column, index}
		<Paper class="column-container">
			<article class="column">
				<div class="column-header">
					{#if column.visible}
						<h2>{column.name}</h2>
					{:else}
						<h2>{column.name} (hidden)</h2>
					{/if}
					<IconButton on:click={() => toggleColumnVisibility(index)}>
						<IconWrapper
							icon={column.visible ? "visibility" : "visibility_off"}
							fillColor="rgb(81, 159, 152)"
						/>
					</IconButton>
				</div>
				{#if column.visible}
					<Content class="content">
						<div class="element-types">
							<ContentCard elementType={column.items} />
						</div>
					</Content>
				{/if}
				<div class="add-button-container">
					{#if column.visible && column.name !== ELEMENT_NAMES.lNode}
						<Button class="add-button" on:click={() => addItemToColumn(index)}>
							+ Create {column.name}
						</Button>
					{/if}
				</div>
		</article>
		</Paper>
	{/each}
</div>
```

Sorry I didn't see that the article tag wasn't at the right spot when we did this together. It should be the wrapper for a column and not the whole columns container (for semantic reasons).

It needs some changes on the css too, I made you suggestion for this.